### PR TITLE
Fix prompt meeting bug

### PR DESCRIPTION
**Pull request by nickoferrall**
This PR resolves the 2 issues in #4238:

1. If you remove a template prompt that is being used in an active/completed meeting, the meeting will crash
2. Removing a prompt removes that prompt from all completed meetings

### Test

- [x] Create a retro meeting and in a second window, remove one of the template prompts without affecting the active retro meeting
- [x] End a retro meeting. Remove a prompt that was used in the completed retro meeting. Revisit the retro meeting by clicking on it from the timeline and see that the prompt was not removed

### CODE REVIEW

File: packages/client/mutations/handlers/handleRemoveReflectTemplatePrompt.ts [hunk 1]
```diff
@@ -1,10 +1,21 @@
 import {RecordSourceSelectorProxy} from 'relay-runtime'
 import pluralizeHandler from './pluralizeHandler'
-import removeFromRefs from '../../utils/relay/removeFromRefs'
+import {IRetrospectiveMeetingSettings, MeetingTypeEnum} from '~/types/graphql'
+import safeRemoveNodeFromArray from '~/utils/relay/safeRemoveNodeFromArray'
 
-const handleRemoveReflectTemplatePrompt = (promptId: string, store: RecordSourceSelectorProxy) => {
-  store.delete(promptId)
-  removeFromRefs(promptId, store, {ReflectTemplate: ['prompts']})
```
Comment (nickoferrall) [inline packages/client/mutations/handlers/handleRemoveReflectTemplatePrompt.ts:7]: `removeFromRefs` was only being used in this file. After removing it here, I've deleted the file

File: packages/server/graphql/types/ReflectPhase.ts [hunk 2]
```diff
@@ -43,9 +43,16 @@ const ReflectPhase = new GraphQLObjectType<any, GQLContext>({
...
+      resolve: async ({meetingId, promptTemplateId}, _args, {dataLoader}) => {
+        const meeting = await dataLoader.get('newMeetings').load(meetingId)
         const prompts = await dataLoader.get('reflectPromptsByTemplateId').load(promptTemplateId)
-        return prompts.filter(({isActive}) => isActive)
+        // only show prompts that were created before the meeting and
+        // either have not been removed or they were removed after the meeting was created
+        return prompts.filter(
+          (prompt) =>
+            prompt.createdAt < meeting.createdAt &&
+            (!prompt.removedAt || meeting.createdAt < prompt.removedAt)
```
Comment (nickoferrall) [inline packages/server/graphql/types/ReflectPhase.ts:54]: The logic only needs to be used here so I felt the custom dataLoader was probably overkill

File: packages/client/mutations/RemoveReflectTemplatePromptMutation.ts [hunk 3]
```diff
@@ -49,10 +50,6 @@ const RemoveReflectTemplatePromptMutation = (
       const payload = store.getRootField('removeReflectTemplatePrompt')
       if (!payload) return
       removeReflectTemplatePromptTeamUpdater(payload as any, {atmosphere, store})
-    },
-    optimisticUpdater: (store) => {
```
Comment (nickoferrall) [inline packages/client/mutations/RemoveReflectTemplatePromptMutation.ts:53]: This `optimisticUpdater` was causing a console err of a store update within a store update. After removing it, I've tested removing a prompt on a 3G network and there doesn't seem to be any difference in the responsiveness

**Comment by nickoferrall** [comment]
Hey @tiffanyhan, to lighten the load on Matt's shoulders, it'd be great if you could review this PR when you get the chance

**Review by tiffanyhan — CHANGES_REQUESTED** [review]
Hey @nickoferrall 

Followed the steps in the AC, and created a meeting with a team template. In a second window, I deleted that template. The active meeting's "reflect" phase then becomes blank, and the facilitator is unable to navigate further. Lmk if you can reproduce it on your side.

Screen Shot 2020-10-07 at 12 39 33 PM

**Comment by nickoferrall** [comment]
@tiffanyhan interesting! I haven't been able to repro that one. To clarify, you get that bug when removing a template prompt or a template? This PR is only focused on fixing the template prompt bugs. 

I've recorded a quick Loom to share with you what I see on my end:
https://www.loom.com/share/4b4a0c32a5184e28b5e2ec7ce7d5ca58

If the error persists for you, it'd be great to get more info on how I can repro it.

**Comment by tiffanyhan** [comment]
@nickoferrall you are totally right. i misread the AC and was removing the entire template rather than a reflect prompt. now it is behaving as expected. the migrations and code looks good to me too. nice work on this!

**Review by tiffanyhan — APPROVED** [review]

**Pull request merged**